calico/datamodel_v1.py

```diff
@@ -220,3 +225,33 @@ def __ne__(self, other):
 
     def __hash__(self):
         return hash(self.endpoint) + hash(self.workload)
+
+
+class TieredPolicyId(object):
+    __slots__ = ["tier", "policy_id"]
+
+    def __init__(self, tier, profile_id):
+        # Intern the strings, which may occur in many IDs.
+        self.tier = intern(tier.encode("utf8"))
+        self.policy_id = intern(profile_id.encode("utf8"))
+
+    def __str__(self):
+        return "%s/%s" % (self.tier, self.policy_id)
+
+    def __repr__(self):
+        return self.__class__.__name__ + ("(%r,%r)" % (self.tier,
+                                                       self.policy_id))
+
+    def __eq__(self, other):
+        if other is self:
+            return True
+        if not isinstance(other, TieredPolicyId):
+            return False
+        return (other.tier == self.tier and
+                other.policy_id == self.policy_id)
+
+    def __ne__(self, other):
+        return not (self == other)
```

Comment:
Seems wrong to me to loop back out into syntactic sugar land - it's not impossible that this could go into an infinite loop (implementation-depending).

How about not self.__eq__(other) instead?